packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -370,7 +370,7 @@ export class Rule extends Resource implements IRule {
         });
         new CfnEventBusPolicy(eventBusPolicyStack, 'GivePermToOtherAccount', {
           action: 'events:PutEvents',
-          statementId: `Allow-account-${sourceAccount}-${this.node.addr}`,
+          statementId: `Allow-account-${sourceAccount}-${this.node.addr}`.substring(0, 64),
```

Comment:
What happens if this cuts these off in a way that there will be a duplicate statementId? Will this cause collisions of any sort?

Reply:
That was my thinking as well, but discussed this on the linked ticket and there should be precedence for this solution.

Reply:
The precedence that @peterwoodworth referred to uses `Names.uniqueResourceName()` to ensure there are no collisions. I don't think he meant that doing it this raw way was the way to go.